src/drivers/stm32f.h

```diff
@@ -16,3 +16,42 @@ struct mg_tcpip_driver_stm32f_data {
 
   uint8_t phy_addr;  // PHY address
 };
+
+#ifndef MG_MAC_ADDRESS
+#define MG_MAC_ADDRESS MG_MAC_ADDRESS_RANDOM
+#endif
+
+#ifndef MG_TCPIP_PHY_ADDR
+#define MG_TCPIP_PHY_ADDR 0
+#endif
+
+#ifndef MG_TCPIP_MDC_CR
+#define MG_TCPIP_MDC_CR 4
+#endif
+
+#define MG_TCPIP_DRIVER_INIT2(mgr)                                 \
+  do {                                                             \
+    struct mg_tcpip_driver_stm32f_data data = {                    \
+        .mdc_cr = MG_TCPIP_MDC_CR,                                 \
+        .phy_adrr = MG_TCPIP_PHY_ADDR,                             \
+    };                                                             \
+    struct mg_tcpip_if mif = {                                     \
+        .mac = MG_MAC_ADDRESS,                                     \
+        .ip = MG_TCPIP_IP,                                         \
+        .mask = MG_TCPIP_MASK,                                     \
+        .gw = MG_TCPIP_GW,                                         \
+        .driver = &mg_tcpip_driver_stm32f,                         \
+        .driver_data = &data,                                      \
+    };                                                             \
+    mg_tcpip_init((mgr), &mif);                                    \
+    MG_INFO(("Driver: stm32fxx, MAC: %M", mg_print_mac, mif.mac)); \
+  } while (0)
+
+#define MG_TCPIP_DRIVER_INIT(mgr)                            \
+  do {                                                       \
+    struct mg_tcpip_if mif = {                               \
+        .mac = MG_MAC_ADDRESS,                               \
+    };                                                       \
+    mg_tcpip_init((mgr), &mif);                              \
+    MG_INFO(("Driver: stm32fxx, MAC: %M", mg_print_mac, 0)); \
+  } while (0)
```

Comment:
To me this is too large and a big chunk will almost be the same for all processors.
I would just add this to net_builtin.c and have small macros with the particulars, as minimum code as possible, maybe static inlines, as this is a header file